pkg/sql/grant_revoke.go

```diff
@@ -185,6 +188,15 @@ func (n *changePrivilegesNode) startExec(params runParams) error {
 	// we update them in KV below.
 	b := p.txn.NewBatch()
 	for _, descriptor := range descriptors {
+		// Disallow privilege changes on system objects. For more context, see #43842.
+		op := "REVOKE"
+		if n.isGrant {
+			op = "GRANT"
+		}
+		if descriptor.GetID() < keys.MinUserDescID {
```

Comment:
I'd prefer if you checked the parent ID

Reply:
Checking ParentID would fail for databases since databaseDesc.GetParentID() always returns 0.